subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -307,7 +321,46 @@ public void useJUnitJupiter(Provider<String> version) {
 
     @Override
     public void useSpock() {
-        useSpock(TestingFramework.SPOCK.getDefaultVersion());
+        useSpock(getDerivedSpockVersion());
+    }
+
+    private static final Set<String> GROOVY_GROUPS = ImmutableSet.of("org.codehaus.groovy", "org.apache.groovy");
+    private static final Set<String> GROOVY_NAMES = ImmutableSet.of("groovy", "groovy-all");
+    private static final Map<VersionNumber, String> GROOVY_TO_SPOCK_VERSION_MAPPING = ImmutableMap.of(
+        VersionNumber.parse("4.0"), "2.2-groovy-4.0",
+        VersionNumber.parse("3.0"), "2.2-groovy-3.0",
+        VersionNumber.parse("2.5"), "2.2-groovy-2.5",
+        VersionNumber.parse("2.4"), "1.3-groovy-2.4",
+        VersionNumber.parse("2.3"), "1.1-groovy-2.3",
+        VersionNumber.parse("2.0"), "1.1-groovy-2.0",
+        VersionNumber.parse("1.8"), "0.7-groovy-1.8",
+        VersionNumber.parse("1.7"), "0.6-groovy-1.7",
+        VersionNumber.parse("1.6"), "0.5-groovy-1.6"
+    );
+
+    private Provider<String> getDerivedSpockVersion() {
+        return getProviderFactory().provider(() -> {
+            Optional<VersionNumber> maybeGroovyJarVersionNumber = runtimeClasspath.getIncoming().getResolutionResult().getAllComponents().stream()
+                .map(ResolvedComponentResult::getModuleVersion)
+                .filter(mvi -> GROOVY_GROUPS.contains(mvi.getGroup()))
+                .filter(mvi -> GROOVY_NAMES.contains(mvi.getName()))
+                .map(ModuleVersionIdentifier::getVersion)
+                .map(VersionNumber::parse)
+                .findFirst();
+
+            // TODO: Should we warn if we can't find Groovy on their classpath?
+            VersionNumber groovyVersion = maybeGroovyJarVersionNumber.orElse(VersionNumber.parse(GroovySystem.getVersion()));
+
+            return getSpockVersionForGroovy(groovyVersion);
+        });
+    }
+
+    private static String getSpockVersionForGroovy(VersionNumber groovyVersion) {
```

Comment:
If we're going to call this in the integration test, we should also unit test this method specifically.